lib/paged-sync.js

```diff
@@ -37,12 +37,15 @@ import mixinStringifySafe from './mixins/stringify-safe'
  * @private
  * @param {Object} http - HTTP client
  * @param {Object} query - Query object
- * @param {boolean} resolveLinks - If links should be resolved
+ * @param {Object} options - If links should be resolved
+ * @param {boolean} [options.resolveLinks = true] - If links should be resolved
+ * @param {boolean} [options.removeUnresolved = false] - If unresolvable links should get removed
+ * @param {boolean} [options.paginate = true] - If links should be resolved
  * @return {Promise<SyncCollection>}
  */
-export default function pagedSync (http, query, { resolveLinks, removeUnresolved }) {
-  if (!query || (!query.initial && !query.nextSyncToken)) {
-    throw new Error('Please provide one of `initial` or `nextSyncToken` parameters for syncing')
+export default function pagedSync (http, query, { resolveLinks = true, removeUnresolved = false, paginate = true } = { resolveLinks: true, removeUnresolved: false, paginate: true }) {
```

Comment:
maybe `options={ resolveLinks: true, removeUnresolved: false, paginate: true }` and inside the function add a check for `undefined`.